metrics_proto.go

```diff
@@ -364,26 +371,23 @@ func labelDescriptorsFromProto(defaults map[string]labelValue, protoLabelKeys []
 	return labelDescriptors
 }
 
-func metricProseFromProto(metric *metricspb.Metric) (name, description, unit string, ok bool) {
-	mname := metric.GetMetricDescriptor().GetName()
-	if mname != "" {
-		name = mname
-		return
-	}
-
+func metricProseFromProto(metric *metricspb.Metric) (name, description, unit string, err error) {
 	md := metric.GetMetricDescriptor()
+	if md == nil {
```

Comment:
https://github.com/census-instrumentation/opencensus-proto/pull/184 reverts the optimization on sending metric name instead of metric descriptor. Thus metric descriptors should always be non-nil.